macros/utils/cross_db_utils/get_profile_creation_query.sql

```diff
@@ -4,16 +4,36 @@
 
 
 {% macro snowflake__get_profile_creation_query(parameters) %}
+{%- set schema_tuples = elementary.get_configured_schemas_from_graph() -%}
+{%- set databases = schema_tuples | map(attribute=0) | unique %}
 CREATE OR REPLACE USER {{ parameters["user"] }} PASSWORD = '{{ parameters["password"] }}';
 CREATE OR REPLACE ROLE {{ parameters["role"] }};
 GRANT ROLE {{ parameters["role"] }} TO USER {{ parameters["user"] }};
 GRANT USAGE ON WAREHOUSE {{ parameters["warehouse"] }} TO ROLE {{ parameters["role"] }};
-GRANT USAGE ON DATABASE {{ parameters["database"] }} TO ROLE {{ parameters["role"] }};
-GRANT USAGE ON SCHEMA {{ parameters["database"] }}.{{ parameters["schema"] }} TO ROLE {{ parameters["role"] }};
+{% for database in databases -%}
+GRANT USAGE,MONITOR ON DATABASE {{ database }} TO ROLE {{ parameters["role"] }};
+{%- endfor %}
+{% for schema_tuple in schema_tuples -%}
+GRANT USAGE,MONITOR ON SCHEMA {{ schema_tuple[0] }}.{{ schema_tuple[1] }} TO ROLE {{ parameters["role"] }};
+{%- endfor %}
+
+-- Read access to elementary schema
 GRANT SELECT ON ALL TABLES IN SCHEMA {{ parameters["database"] }}.{{ parameters["schema"] }} TO ROLE {{ parameters["role"] }};
 GRANT SELECT ON FUTURE TABLES IN SCHEMA {{ parameters["database"] }}.{{ parameters["schema"] }} TO ROLE {{ parameters["role"] }};
 GRANT SELECT ON ALL VIEWS IN SCHEMA {{ parameters["database"] }}.{{ parameters["schema"] }} TO ROLE {{ parameters["role"] }};
 GRANT SELECT ON FUTURE VIEWS IN SCHEMA {{ parameters["database"] }}.{{ parameters["schema"] }} TO ROLE {{ parameters["role"] }};
+
+-- Metadata access to rest of the schemas used within the dbt project
+{% for database, schema in schema_tuples -%}
+GRANT REFERENCES ON ALL TABLES IN SCHEMA {{ database }}.{{ schema }} TO ROLE {{ parameters["role"] }};
+GRANT REFERENCES ON FUTURE TABLES IN SCHEMA {{ database }}.{{ schema }} TO ROLE {{ parameters["role"] }};
+GRANT REFERENCES ON ALL VIEWS IN SCHEMA {{ database }}.{{ schema }} TO ROLE {{ parameters["role"] }};
+GRANT REFERENCES ON FUTURE VIEWS IN SCHEMA {{ database }}.{{ schema }} TO ROLE {{ parameters["role"] }};
+{% endfor -%}
+
+-- Needed for access to query history
+GRANT DATABASE ROLE SNOWFLAKE.USAGE_VIEWER TO ROLE {{ parameters["role"] }};
```

Comment:
Are you sure these roles exist when snowflake is not enterprise?

Reply:
yes definitely
our test account is not enterprise